File: token-client-spring/src/test/java/no/nav/security/token/support/client/spring/oauth2/ClientConfigurationPropertiesTestWithResourceUrl.java
package no.nav.security.token.support.client.spring.oauth2;

import no.nav.security.token.support.client.core.ClientAuthenticationProperties;
import no.nav.security.token.support.client.core.ClientProperties;
import no.nav.security.token.support.client.spring.ClientConfigurationProperties;
import no.nav.security.token.support.core.context.TokenValidationContextHolder;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.MockitoAnnotations;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.web.client.RestTemplateAutoConfiguration;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.test.context.ActiveProfiles;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.*;

@SpringBootTest(classes = {OAuth2ClientConfiguration.class, ConfigurationWithCacheEnabled.class})
@ActiveProfiles("test-withresourceurl")
class ClientConfigurationPropertiesTestWithResourceUrl {

    @MockBean
    private TokenValidationContextHolder tokenValidationContextHolder;

    @Autowired
    private ClientConfigurationProperties clientConfigurationProperties;

    @BeforeEach
    void before() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    void testClientConfigIsValid() {
        assertThat(clientConfigurationProperties).isNotNull();
        assertThat(clientConfigurationProperties.getRegistration()).isNotNull();
        ClientProperties clientProperties =
            clientConfigurationProperties.getRegistration().values().stream().findFirst().orElse(null);
        assertThat(clientProperties).isNotNull();
        ClientAuthenticationProperties auth = clientProperties.getAuthentication();
        assertThat(auth).isNotNull();
        assertThat(auth.getClientAuthMethod()).isNotNull();
        assertThat(auth.getClientId()).isNotNull();
        assertThat(auth.getClientSecret()).isNotNull();
        assertThat(clientProperties.getScope()).isNotEmpty();
        assertThat(clientProperties.getTokenEndpointUrl()).isNotNull();
        assertThat(clientProperties.getGrantType().getValue()).isNotNull();
        assertThat(clientProperties.getResourceUrl()).isNotNull();
    }
}

File: token-client-spring/src/test/kotlin/no/nav/security/token/support/client/spring/oauth2/ClientConfigurationPropertiesTestWithResourceUrl.kt
package no.nav.security.token.support.client.spring.oauth2


import no.nav.security.token.support.client.spring.ClientConfigurationProperties
import no.nav.security.token.support.core.context.TokenValidationContextHolder
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.mockito.junit.jupiter.MockitoExtension
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.restclient.autoconfigure.RestClientAutoConfiguration
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.bean.override.mockito.MockitoBean

@SpringBootTest(classes = [OAuth2ClientConfiguration::class, RestClientAutoConfiguration::class])
@ExtendWith(MockitoExtension::class)
@ActiveProfiles("test-withresourceurl")
internal class ClientConfigurationPropertiesTestWithResourceUrl {

    private  val matcher = object: ClientConfigurationPropertiesMatcher {}
    @MockitoBean
    private val tokenValidationContextHolder: TokenValidationContextHolder? = null

    @Autowired
    private lateinit var clientConfigurationProperties: ClientConfigurationProperties

    @Test
    fun testClientConfigIsValid() {
        assertThat(matcher.findProperties(clientConfigurationProperties, "https://isdialogmelding.dev.intern.nav.no/api/person/v1/behandler/self")).isNotNull
        assertThat(clientConfigurationProperties).isNotNull
        val clientProperties = clientConfigurationProperties.registration.values.firstOrNull()
        assertThat(clientProperties).isNotNull
        val auth = clientProperties?.authentication
        assertThat(auth?.clientId).isNotNull
        assertThat(auth?.clientSecret).isNotNull
        assertThat(clientProperties?.scope).isNotEmpty
        assertThat(clientProperties?.tokenEndpointUrl).isNotNull
        assertThat(clientProperties?.grantType?.value).isNotNull
        assertThat(clientProperties?.resourceUrl).isNotNull
    }
}